components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -459,6 +459,35 @@ public String getPublishedDefaultVersion(APIIdentifier apiid) throws APIManageme
 
 
     /**
+     * This method is used to save the wsdl file in the registry
+     * This is used when user starts api creation with a soap endpoint
+     *
+     * @param api api object
+     * @throws APIManagementException
+     * @throws RegistryException
+     */
+    private void updateWsdl(API api) throws APIManagementException, RegistryException {
+
+        registry.beginTransaction();
```

Comment:
Do we have any rollback option as part of registry transaction lifecyle? If so, shouldn't that be used to clean up all allocated transactional resources when some error is thrown?